urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/common/number/IntervalType.java

```diff
@@ -0,0 +1,280 @@
+		if(a.type()==b.type())
+			return new IntervalNumber[]{a,b};
+		IntervalType t = a.type().promote(b.type());
+		return new IntervalNumber[]{promote(a,t),promote(b,t)};
+	}
+	public static IntervalNumber[] promote(IntervalNumber a, ScaleNumber b){
+		return promote(a,promote(b));
+	}
+	public static IntervalNumber[] promote(IntervalNumber a, Number b){
+		return promote(a, promote(b));
+	}
+
+	public static IntervalNumber promoteOp(IntervalNumber n1, IntervalNumber n2, CohoNumber.ArithOp op){
+		IntervalNumber[] operands = promote(n1,n2);
+		IntervalType t = operands[0].type();
+		if(t == IntegerInterval.type){
+			IntegerInterval op1 = (IntegerInterval)operands[0];
+			IntegerInterval op2 = (IntegerInterval)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return IntegerInterval.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}
+		}else if(t == LongInterval.type){
+			LongInterval op1 = (LongInterval)operands[0];
+			LongInterval op2 = (LongInterval)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return IntegerInterval.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else if(t == DoubleInterval.type){
+			DoubleInterval op1 = (DoubleInterval)operands[0];
+			DoubleInterval op2 = (DoubleInterval)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return IntegerInterval.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else if(t == APRInterval.type){
+			APRInterval op1 = (APRInterval)operands[0];
+			APRInterval op2 = (APRInterval)operands[1];
+			switch(op){
+			case ADD:
+				return op1.add(op2);
+			case SUB:
+				return op1.sub(op2);
+			case MULT:
+				return op1.mult(op2);
+			case DIV:
+				return op1.div(op2);
+			case MAX:
+				return op1.max(op2);
+			case MIN:
+				return op1.min(op2);
+			case CMP:
+				return IntegerInterval.create(op1.compareTo(op2));
+			default:
+				throw new UnsupportedOperationException(op.name());
+			}			
+		}else
+			throw new RuntimeException("Unknow IntervalType"+t);
+	}
+	public static IntervalNumber promoteOp(IntervalNumber n1, Number n2, CohoNumber.ArithOp op){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 84 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.